pydra/engine/core.py

```diff
@@ -534,7 +548,11 @@ def done(self):
                 return True
         else:
             if self.result():
```

Comment:
since result is a function, shouldn't it be assigned to a variable so that it is not recomputed?